cli/tests/qa/test_semgrep_rules_repo.py

```diff
@@ -39,34 +29,26 @@ def in_semgrep_rules_repo(tmpdir_factory):
 
 
 @pytest.mark.slow
-def test_semgrep_rules_repo__test(in_semgrep_rules_repo):
-    runner = CliRunner()
-    results = runner.invoke(
-        cli,
-        [
-            "--disable-version-check",
-            "--metrics=off",
-            "--strict",
-            "--test",
-            "--test-ignore-todo",
-        ],
-    )
-    print(results.output)
-    assert results.exit_code == 0
+def test_semgrep_rules_repo__validate(in_semgrep_rules_repo):
+    """Validate the rule files found in the semgrep-rules repo.
+
+    This runs the same tests as semgrep-rules' own CI does to avoid
+    diverging expectations.
+    """
+    p = subprocess.Popen(["make", "validate"])
+    p.communicate()
+    exit_code = p.wait()
+    assert exit_code == 0
 
 
 @pytest.mark.slow
-def test_semgrep_rules_repo__validate(in_semgrep_rules_repo):
-    runner = CliRunner()
-    results = runner.invoke(
-        cli,
-        [
-            "--disable-version-check",
-            "--metrics=off",
-            "--strict",
-            "--validate",
-            "--config=.",
-        ],
-    )
-    print(results.output)
-    assert results.exit_code == 0
+def test_semgrep_rules_repo__test(in_semgrep_rules_repo):
+    """Test the pairs rule/target found in the semgrep-rules repo.
+
+    This runs the same tests as semgrep-rules' own CI does to avoid
+    diverging expectations.
+    """
+    p = subprocess.Popen(["make", "test-only"])
```

Comment:
`runner.invoke` was used because it allows you to run tests via a debugger and step all the way down to the `semgrep` module's code. Creating a new subprocess escapes any debugger you attach to the test.

Using a subprocess also adds a bit more overhead, but that's probably fine since this is a long-running test anyway.

Please merge only if you're comfortable with the debuggability tradeoff.

Reply:
Good to know, thanks for sharing. We're now depending on the invocation decided by the semgrep-rules submodule, so it's simpler this way.